TheForceEngine/TFE_DarkForces/Actor/actor.cpp

```diff
@@ -1012,23 +1013,40 @@ namespace TFE_DarkForces
 				SecObject* projObj = proj->logic.obj;
 				projObj->yaw = obj->yaw;
 				
-				// Handle x and z fire offset.
-				if (attackMod->fireOffset.x | attackMod->fireOffset.z)
+				if (attackMod->projType == PROJ_THERMAL_DET || attackMod->projType == PROJ_MORTAR)
```

Comment:
Does this change vanilla behavior? If so it will need to be made optional.

Reply:
It does not change vanilla behaviour, because none of the vanilla characters have thermal detonators as their primary attack (`case STATE_FIRE1:`)

To  clarify the change I've made here:
The original code is found on lines 1033-1049, exactly as it was before, but now within the `else` code block. So it now follows the same logic as occurs with `STATE_FIRE2`

`if (arcing projectile)
{
     aimArcing
}
else
{
    aimAtTarget
}`

Vanilla enemies will all fall through to the `else` block (i.e. the original code) because none of them use an arcing projectile as their `STATE_FIRE1`. 

Hope I've explained that adequately

Reply:
Ok, thanks for the explanation - it makes sense now.